rcl_yaml_param_parser/test/test_parse_yaml.cpp

```diff
@@ -497,6 +497,54 @@ TEST(test_file_parser, maximum_number_parameters) {
   // No cleanup, rcl_parse_yaml_file takes care of that if it fails.
 }
 
+TEST(test_file_parser, special_float_point) {
+  rcutils_reset_error();
+  EXPECT_TRUE(rcutils_get_cwd(cur_dir, 1024)) << rcutils_get_error_string().str;
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  char * test_path = rcutils_join_path(cur_dir, "test", allocator);
+  ASSERT_TRUE(NULL != test_path) << rcutils_get_error_string().str;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    allocator.deallocate(test_path, allocator.state);
+  });
+
+  // Test special float point with correct value.
+  {
+    char * path = rcutils_join_path(test_path, "special_float1.yaml", allocator);
+    ASSERT_TRUE(NULL != path) << rcutils_get_error_string().str;
+    OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+    {
+      allocator.deallocate(path, allocator.state);
+    });
+    ASSERT_TRUE(rcutils_exists(path)) << "No test YAML file found at " << path;
+    rcl_params_t * params_hdl = rcl_yaml_node_struct_init(allocator);
+    ASSERT_TRUE(NULL != params_hdl) << rcutils_get_error_string().str;
+    OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+    {
+      rcl_yaml_node_struct_fini(params_hdl);
+    });
+    bool res = rcl_parse_yaml_file(path, params_hdl);
+    EXPECT_TRUE(res) << rcutils_get_error_string().str;
+  }
+
+  // Test special float point with improperly value.
+  {
+    char * path = rcutils_join_path(test_path, "special_float2.yaml", allocator);
+    ASSERT_TRUE(NULL != path) << rcutils_get_error_string().str;
+    OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+    {
+      allocator.deallocate(path, allocator.state);
+    });
+    ASSERT_TRUE(rcutils_exists(path)) << "No test YAML file found at " << path;
+    rcl_params_t * params_hdl = rcl_yaml_node_struct_init(allocator);
+    ASSERT_TRUE(NULL != params_hdl) << rcutils_get_error_string().str;
+    bool res = rcl_parse_yaml_file(path, params_hdl);
+    fprintf(stderr, "%s\n", rcutils_get_error_string().str);
+    EXPECT_FALSE(res);
```

Comment:
Hm, do we really expect this to fail?  Shouldn't this succeed, but make the parameter a string rather than a double?